src/adzerk/response_models.rs

```diff
@@ -284,9 +284,87 @@ fn tracking_url_to_shim(url: String) -> Result<String, ProxyError> {
 #[cfg(test)]
 mod tests {
     use super::Decision;
+    use crate::adzerk::response_models::{
+        clean_sponsored_by_override, get_cdn_image, get_is_video, get_personalization_models,
+        tracking_url_to_shim,
+    };
+    use std::collections::HashMap;
 
     #[test]
     fn test_deserialize_decisions() {
         let _: Vec<Decision> = serde_json::from_str(include_str!("mock_decision.json")).unwrap();
     }
+
+    #[test]
+    fn test_tracking_url_to_shim() {
+        let test_string: String = "https://example.local/r?e=123&s=456&j=789".to_string();
+        let test_result = tracking_url_to_shim(test_string).unwrap();
+        assert_eq!(test_result, "0,123,456")
+    }
+
+    #[test]
+    fn test_is_video() {
+        let test_map: HashMap<Option<String>, Option<bool>> = [
+            (Some(String::from("t")), Some(true)),
+            (Some(String::from("off")), Some(false)),
+            (Some(String::from("1")), Some(true)),
+        ]
+        .iter()
+        .cloned()
+        .collect();
+
+        for (key, value) in test_map {
+            assert_eq!(get_is_video(key), value);
+        }
```

Comment:
In more recent versions of Rust, a `HashMap` can be build using
```rust
HashMap::from_iter([(key0, value0), (key1, value1), ...])
```
The older `.iter().cloned().collect()` pattern is no longer needed, since [there is a proper `IntoIterator` implementation for arrays now](https://doc.rust-lang.org/edition-guide/rust-2021/IntoIterator-for-arrays.html).

That said, there isn't really a reason to build a `HashMap` her anyway. A simpler implementation could look like this:
```suggestion
        let test_cases = [("t", true), ("off", false), ("1", true)];
        for (input, output) in test_cases {
            assert_eq!(get_is_video(Some(input.into())), Some(output));
        }
```

Reply:
Oh that's a much better way to create a HashMap, thanks @smarnach